tests/unit/modules/test_nxos_upgrade.py

```diff
@@ -0,0 +1,366 @@
+# -*- coding: utf-8 -*-
+'''
+    :codeauthor: Thomas Stoner <tmstoner@cisco.com>
+'''
+
+# Copyright (c) 2018 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Import Python libs
+from __future__ import absolute_import, print_function, unicode_literals
+
+# Import Salt Testing Libs
+from tests.support.mixins import LoaderModuleMockMixin
+from tests.support.unit import skipIf, TestCase
+from tests.support.mock import (
+    NO_MOCK,
+    NO_MOCK_REASON,
+    MagicMock,
+    patch)
+
+from tests.unit.modules.nxos.nxos_n95k import N95KPlatform
+from tests.unit.modules.nxos.nxos_n93k import N93KPlatform
+from tests.unit.modules.nxos.nxos_n93klxc import N93KLXCPlatform
+from tests.unit.modules.nxos.nxos_n7k import N7KPlatform
+from tests.unit.modules.nxos.nxos_n5k import N5KPlatform
+from tests.unit.modules.nxos.nxos_n36k import N36KPlatform
+from tests.unit.modules.nxos.nxos_n3k import N3KPlatform
+
+from salt.exceptions import (NxosError, CommandExecutionError)
+
+# Import Salt Libs
+import salt.modules.nxos_upgrade as nxos_upgrade
+
+# pylint: disable-msg=C0103
+# pylint: disable-msg=C0301
+# pylint: disable-msg=E1101
+# pylint: disable-msg=R0904
+
+
+@skipIf(NO_MOCK, NO_MOCK_REASON)
+class NxosUpgradeTestCase(TestCase, LoaderModuleMockMixin):
+
+    """ Test cases for salt.modules.nxos_upgrade """
+
+    platform_list = None
+
+    @staticmethod
+    def assert_platform_upgrade(condition, platform):
+
+        """ Assert platform upgrade condition and display appropriate chassis & images upon assertion failure """
+
+        assert bool(condition), "{0}: Upgrade {1} -> {2}".format(platform.chassis, platform.cimage, platform.nimage)
```

Comment:
This is how the module information refers to the the current image and the new image.  nimage is 'new image'.  Because of this I'd prefer to stick to the same name and just add a comment.